src/perform_step/extrapolation_perform_step.jl

```diff
@@ -230,6 +230,193 @@ function perform_step!(integrator,cache::AitkenNevilleConstantCache,repeat_step=
   integrator.k[2] = integrator.fsallast
 end
 
+function initialize!(integrator,cache::ImplicitEulerExtrapolationCache)
+  integrator.kshortsize = 2
+  @unpack k,fsalfirst = cache
+  integrator.fsalfirst = fsalfirst
+  integrator.fsallast = k
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst,integrator.uprev,integrator.p,integrator.t) # For the interpolation, needs k at the updated point
+  integrator.destats.nf += 1
+
+  cache.step_no = 1
+  cache.cur_order = max(integrator.alg.init_order, integrator.alg.min_order)
+end
+
+function perform_step!(integrator,cache::ImplicitEulerExtrapolationCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack k,T,utilde,atmp,dtpropose,cur_order,A = cache
+  @unpack u_tmps, k_tmps, nlsolver, z = cache
+
+  alg = unwrap_alg(integrator, true)
+
+  max_order = min(size(T)[1],cur_order+1)
+
+  for i in 1:max_order
+    dt_temp = dt/(2^(i-1)) # Romberg sequence
+    u_tmp = copy(uprev)
```

Comment:
use a temporary from cache for this

Reply:
cache optimization still needs to be done. I am trying to converge this method for order > 1 for now.

Reply:
oh okay, no worries